LiteDB/Client/Shared/SharedEngine.cs

```diff
@@ -15,7 +15,7 @@ namespace LiteDB
     public class SharedEngine : ILiteEngine
     {
         private readonly EngineSettings _settings;
-        private readonly Mutex _mutex;
+        private readonly Semaphore _semaphore;
```

Comment:
This is not part of the issue. Revert this please.

Reply:
reverted